service-front/module/Application/src/Model/Service/Lpa/ContinuationSheets.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Application\Model\Service\Lpa;
+
+use Application\Model\Service\AbstractService;
+use Opg\Lpa\DataModel\Lpa\Formatter as LpaFormatter;
+use Opg\Lpa\DataModel\Lpa\Lpa;
+
+class ContinuationSheets
+{
+    /**
+     * Gathers an array on conditions where continuation sheet(s) would be generated in the PDF.
+     *
+     * @param Lpa $lpa
+     * @return array
+     */
+    public function getContinuationNoteKeys(Lpa $lpa): array
+    {
+        //Array of keys to know which extra notes to show in template for continuation sheets
+        $continuationNoteKeys = array();
+        $extraBlockPeople = null;
+        $paCount = count($lpa->document->primaryAttorneys);
+        $raCount = count($lpa->document->replacementAttorneys);
+        $pnCount = count($lpa->document->peopleToNotify);
+
+        if ($paCount > 4 && $raCount > 2 && $pnCount > 4) {
+            $extraBlockPeople = 'ALL_PEOPLE_OVERFLOW';
+        } elseif ($paCount > 4 && $raCount > 2) {
+            $extraBlockPeople =  'ALL_ATTORNEY_OVERFLOW';
+        } elseif ($paCount > 4 && $pnCount > 4) {
+            $extraBlockPeople =  'PRIMARY_ATTORNEY_AND_NOTIFY_OVERFLOW';
+        } elseif ($raCount > 2 &&  $pnCount > 4) {
+            $extraBlockPeople =  'REPLACEMENT_ATTORNEY_AND_NOTIFY_OVERFLOW';
+        } elseif ($paCount > 4) {
+            $extraBlockPeople =  'PRIMARY_ATTORNEY_OVERFLOW';
+        } elseif ($raCount > 2) {
+            $extraBlockPeople =  'REPLACEMENT_ATTORNEY_OVERFLOW';
+        } elseif ($pnCount > 4) {
+            $extraBlockPeople =  'NOTIFY_OVERFLOW';
+        }
+
+        if ($extraBlockPeople != null) {
+            array_push($continuationNoteKeys, $extraBlockPeople);
+        }
+
+        if ($paCount > 4 || $raCount > 2 || $pnCount > 4) {
+            array_push($continuationNoteKeys, 'ANY_PEOPLE_OVERFLOW');
+        }
+
+        if (
+            isset($lpa->document->primaryAttorneyDecisions->howDetails) ||
+            isset($lpa->document->replacementAttorneyDecisions->howDetails) ||
+            isset($lpa->document->replacementAttorneyDecisions->when)
+        ) {
+            array_push($continuationNoteKeys, 'HAS_ATTORNEY_DECISIONS');
+        }
+
+        if (isset($lpa->document->donor)) {
+            if (!$lpa->document->donor->canSign) {
+                array_push($continuationNoteKeys, 'CANT_SIGN');
+            }
+        }
+
+        $someAttorneyIsTrustCorp = false;
```

Comment:
Probably don't need this variable. Instead, I'd just loop over the two arrays added together, then do `array_push($continuationNoteKeys, 'HAS_TRUST_CORP');` if an attorney is a trust corp and break the for loop, e.g.

```
$allAttorneys = array_merge($lpa->document->primaryAttorneys, $lpa->document->replacementAttorneys);
foreach ($allAttorneys as $attorney) {
    if (isset($attorney->number)) {
        $continuationNoteKeys[] = 'HAS_TRUST_CORP';
        break;
    }
}
```

This would replace lines 64-80.